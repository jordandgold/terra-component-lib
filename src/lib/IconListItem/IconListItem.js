import React from "react";
import "./IconListItem.scss";
import PropTypes from "prop-types";
import Icon from "../Icon/Icon";

const IconListItem = ({ content }) => {
  return (
    <article className="ter-icon-list-item">
      <div className="ter-icon-list-item--icon-wrapper">
        <Icon type="enclosed-arrow-right-dark-32px" size="32px" />
      </div>
      <div className="ter-icon-list-item--content-wrapper">
        <h4 className="ter-icon-list-item--title">{content.title}</h4>
        <p className="ter-icon-list-item--body">{content.body}</p>
      </div>
    </article>
  );
};

export default IconListItem;

IconListItem.propTypes = {
  content: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired
  })
};
