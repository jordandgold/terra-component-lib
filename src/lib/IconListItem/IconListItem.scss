@import "../global-styles/terra.scss";

.ter-icon-list-item {
  display: flex;
  max-width: 500px;
  padding: 16px;

  &--icon-wrapper {
    margin-right: 16px;
  }

  &--title {
    font-size: font-size("md");
  }
}
