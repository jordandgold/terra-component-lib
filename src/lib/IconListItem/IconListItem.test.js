import React from "react";
import IconListItem from "./IconListItem";
import { shallow } from "enzyme";
import { isIterable } from "core-js";

describe("IconListItem", () => {
  let wrapper;

  beforeEach(() => {
    const mockContent = {
      title: "Design is good. Period.",
      body:
        "Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!"
    };
    wrapper = shallow(<IconListItem content={mockContent} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
