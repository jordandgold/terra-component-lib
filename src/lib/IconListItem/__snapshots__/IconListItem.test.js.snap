// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconListItem should match the snapshot 1`] = `
<article
  className="ter-icon-list-item"
>
  <div
    className="ter-icon-list-item--icon-wrapper"
  >
    <Icon
      size="32px"
      type="enclosed-arrow-right-dark-32px"
    />
  </div>
  <div
    className="ter-icon-list-item--content-wrapper"
  >
    <h4
      className="ter-icon-list-item--title"
    >
      Design is good. Period.
    </h4>
    <p
      className="ter-icon-list-item--body"
    >
      Forget the fat lady! You're obsessed with the fat lady! Drive us out of here!
    </p>
  </div>
</article>
`;
